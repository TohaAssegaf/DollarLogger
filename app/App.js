import Navigation from './components/navigation/Navigation'
import store from './store'
import React from 'react'
import { Provider } from 'react-redux'

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Navigation />
      </Provider>
    );
  }
}
