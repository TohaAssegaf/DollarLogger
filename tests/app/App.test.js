import React from 'react';
import App from '~/app/App';

import renderer from 'react-test-renderer';
//import mocked RNFirebase
import RNFirebase from 'react-native-firebase'

//reset mocks
beforeEach(() => {
  RNFirebase.reset()
})

//set mocked data
const ref = RNFirebase.firebase.database().ref('users/')
ref.data = users

//verify listener called
expect(ref.on.mock.calls.length).toBe(1)

it('renders without crashing', () => {
  const rendered = renderer.create( <App /> ).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders correctly', () => {
  const rendered = renderer.create( <App /> ).toJSON();
  expect(rendered).toMatchSnapshot();
});
