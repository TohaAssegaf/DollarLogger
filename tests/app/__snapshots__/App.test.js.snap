// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#EFEFF4",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Loading...
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
